include/pgduckdb/pgduckdb_types.hpp

```diff
@@ -1,25 +1,20 @@
 #pragma once
 
-#include "duckdb.hpp"
-#include "duckdb/common/shared_ptr.hpp"
-#include "duckdb/common/extra_type_info.hpp"
-
-extern "C" {
-#include "postgres.h"
-#include "executor/tuptable.h"
-}
-
-#include "pgduckdb/scan/postgres_scan.hpp"
+#include "pgduckdb/pg/declarations.hpp"
 
 namespace pgduckdb {
 
+class PostgresScanGlobalState;
+class PostgresScanLocalState;
+typedef uint64_t idx_t; // From duckdb.h
+
 // DuckDB has date starting from 1/1/1970 while PG starts from 1/1/2000
 constexpr int32_t PGDUCKDB_DUCK_DATE_OFFSET = 10957;
-constexpr int64_t PGDUCKDB_DUCK_TIMESTAMP_OFFSET = INT64CONST(10957) * USECS_PER_DAY;
+constexpr int64_t PGDUCKDB_DUCK_TIMESTAMP_OFFSET = static_cast<int64_t>(10957) * static_cast<int64_t>(86400000000) /* USECS_PER_DAY */;
```

Comment:
Seems it would be nice to reuse our constants
```suggestion
constexpr int64_t PGDUCKDB_DUCK_TIMESTAMP_OFFSET = static_cast<int64_t>(PGDUCKDB_DUCK_DATE_OFFSET) * static_cast<int64_t>(86400000000) /* USECS_PER_DAY */;
```